src/scheduler/gc_work.rs

```diff
@@ -600,9 +614,9 @@ pub trait ProcessEdgesWork:
         }
     }
 
-    /// Process an edge, including loading the object reference from the memory slot,
+    /// Process a slot, including loading the object reference from the memory slot,
     /// trace the object and store back the new object reference if necessary.
-    fn process_edge(&mut self, slot: EdgeOf<Self>) {
+    fn process_slot(&mut self, slot: SlotOf<Self>) {
```

Comment:
I don't want to name it `process_edge` because there are two ways to use `ProcessEdgesWork`.

1.  Put slots in it and process each of them.  That what `process_slot` and `process_slots` does.
2.  Use the `trace_object` method directly.  We use it in the following way:

```rust
// <E: ProcessEdgesWork>
let w = E::new(vec![], ...); // Create a concrete E without any slots
w.set_worker(worker);
for obj in /* go through a list of VM objects or special data structures such as finalization table */ {
  let objref = obj.field1;
  let newref = w.trace_object(objref); // Call `trace_object` to trace an edge
  obj.field1 = newref;
  // visit other fields
}
w.flush(); // Flush newly visited objects into `ScanObjects`
```

We have the second use pattern in `Scanning::scan_object_and_trace_edges` (via `ProcessEdgeWorkTracer` as an impl of `ObjectTracer`), `SoftRefProcessing`, `Finalization` and `Scanning::process_weak_ref` (also using `ObjectTracer`).  The semantics is still processing edges, but without using the `Slot` trait.

Both (1) and (2) are processing edges.  It is true that the method `process_edge` also processes edges, but I want to rename it to emphasize that it is not the only way to process edge, as it only processes edges represented as slots.

Reply:
OK. This makes sense.